Task2/service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImplementation.java

```diff
@@ -0,0 +1,97 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.dto.CertificateFilterDto;
+import com.epam.esm.entity.dto.CreateGiftCertificateDto;
+import com.epam.esm.entity.dto.UpdateGiftCertificateDto;
+import com.epam.esm.repository.GiftCertificateRepository;
+import com.epam.esm.repository.TransactionManager;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.validator.GiftCertificateValidator;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+public class GiftCertificateServiceImplementation implements GiftCertificateService {
+    private GiftCertificateRepository giftCertificateRepository;
+    private TransactionManager transactionManager;
+    private GiftCertificateValidator giftCertificateValidator;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Autowired
+    public void setGiftCertificateRepository(GiftCertificateRepository giftCertificateRepository) {
+        this.giftCertificateRepository = giftCertificateRepository;
+    }
+
+    @Autowired
+    public void setTransactionManager(TransactionManager transactionManager) {
+        this.transactionManager = transactionManager;
+    }
+
+    @Autowired
+    public void setGiftCertificateValidator(GiftCertificateValidator giftCertificateValidator) {
+        this.giftCertificateValidator = giftCertificateValidator;
+    }
```

Comment:
What is the purpose of injecting in setters? Why do not you use constructor injection as in other places?

Reply:
I usually use setter injection, just find it more comfortable. I used constructor injection in the repository class as it required a constructor injection for the data source. In case constructor injection is more preferable overall I can change that in my projects